packages/grid/x-data-grid-pro/src/tests/columns.DataGridPro.test.tsx

```diff
@@ -420,4 +421,18 @@ describe('<DataGridPro /> - Columns', () => {
       ]);
     });
   });
+
+  it('should throw if unsupported column type is used', () => {
+    const singleColumns = [{ field: 'brand', type: 'str' }];
+
+    expect(() => {
+      render(<Test columns={singleColumns} />);
+      // @ts-expect-error need to migrate helpers to TypeScript
```

Comment:
Could you merge with `master` first. We don't need to skip these TypeScript errors anymore.